### ISSUE TITLE
BindingUsageController application is continuously restarted because of panic

### ISSUE THREAD

**Issue by mszostok** [kyma-project/kyma#2011]
**Description**

The BindingUsageController was restarted 45 times over 19h on **nightly** cluster.

**Troubleshooting**

 Restarts are caused by  **"Observed a panic: "invalid memory address or nil pointer dereference"**.  The problem is that we have an unchecked error while getting SBU via client-go, see: 
https://github.com/kyma-project/kyma/blob/master/components/binding-usage-controller/internal/controller/controller.go#L236

Error log:
```
{"level":"debug","log":{"message":"Error processing \"ui-api-acceptance/binding-usage-test\" (will retry - it's 0 of 15): while processing ServiceBindingUsage \"ui-api-acceptance/binding-usage-test\": Reason: EnsureLabelsAppliedError, details: while ensuring proper labels on kind deployment: while applying labels: while ensuring labels on \"deployment\" \"sample-deployment\" in namespace \"ui-api-acceptance\": while getting resource: deployments.apps \"sample-deployment\" not found","service":"controller:service-binding-usage","time":"2018-12-11T21:20:52.914Z"}}
{"level":"info","log":{"message":"Shutting down service binding usage controller","service":"controller:service-binding-usage","time":"2018-12-11T21:20:52.917Z"}}
ERROR: logging before flag.Parse: E1211 21:20:52.917426       1 runtime.go:66] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
/home/jenkins/workspace/kyma/components/binding-usage-controller/src/github.com/kyma-project/kyma/components/binding-usage-controller/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:72
/home/jenkins/workspace/kyma/components/binding-usage-controller/src/github.com/kyma-project/kyma/components/binding-usage-controller/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:65
/home/jenkins/workspace/kyma/components/binding-usage-controller/src/github.com/kyma-project/kyma/components/binding-usage-controller/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:51
/usr/local/go/src/runtime/asm_amd64.s:573
/usr/local/go/src/runtime/panic.go:502
/usr/local/go/src/runtime/panic.go:63
/usr/local/go/src/runtime/signal_unix.go:388
/home/jenkins/workspace/kyma/components/binding-usage-controller/src/github.com/kyma-project/kyma/components/binding-usage-controller/internal/controller/controller.go:529
/home/jenkins/workspace/kyma/components/binding-usage-controller/src/github.com/kyma-project/kyma/components/binding-usage-controller/internal/controller/controller.go:238
/home/jenkins/workspace/kyma/components/binding-usage-controller/src/github.com/kyma-project/kyma/components/binding-usage-controller/internal/controller/controller.go:195
/home/jenkins/workspace/kyma/components/binding-usage-controller/src/github.com/kyma-project/kyma/components/binding-usage-controller/internal/controller/controller.go:191
/home/jenkins/workspace/kyma/components/binding-usage-controller/src/github.com/kyma-project/kyma/components/binding-usage-controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133
/home/jenkins/workspace/kyma/components/binding-usage-controller/src/github.com/kyma-project/kyma/components/binding-usage-controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134
/home/jenkins/workspace/kyma/components/binding-usage-controller/src/github.com/kyma-project/kyma/components/binding-usage-controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
/home/jenkins/workspace/kyma/components/binding-usage-controller/src/github.com/kyma-project/kyma/components/binding-usage-controller/internal/controller/controller.go:191
/home/jenkins/workspace/kyma/components/binding-usage-controller/src/github.com/kyma-project/kyma/components/binding-usage-controller/cmd/controller/main.go:120
/usr/local/go/src/runtime/proc.go:198
/usr/local/go/src/runtime/asm_amd64.s:2361
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xe1c04d]

goroutine 1 [running]:
github.com/kyma-project/kyma/components/binding-usage-controller/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/home/jenkins/workspace/kyma/components/binding-usage-controller/src/github.com/kyma-project/kyma/components/binding-usage-controller/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58 +0x107
panic(0xfbfa60, 0x19d8160)
	/usr/local/go/src/runtime/panic.go:502 +0x229
github.com/kyma-project/kyma/components/binding-usage-controller/internal/controller.(*ServiceBindingUsageController).updateStatus(0xc4203e8000, 0x0, 0x1108cfc, 0x5, 0x1108c48, 0x5, 0xbefc286d36acff9b, 0x26ebf505681, 0x19ea020, 0xbefc286d36acffff, ...)
	/home/jenkins/workspace/kyma/components/binding-usage-controller/src/github.com/kyma-project/kyma/components/binding-usage-controller/internal/controller/controller.go:529 +0x4d
github.com/kyma-project/kyma/components/binding-usage-controller/internal/controller.(*ServiceBindingUsageController).processNextWorkItem(0xc4203e8000, 0xc42030fa00)
	/home/jenkins/workspace/kyma/components/binding-usage-controller/src/github.com/kyma-project/kyma/components/binding-usage-controller/internal/controller/controller.go:238 +0x754
github.com/kyma-project/kyma/components/binding-usage-controller/internal/controller.(*ServiceBindingUsageController).worker(0xc4203e8000)
	/home/jenkins/workspace/kyma/components/binding-usage-controller/src/github.com/kyma-project/kyma/components/binding-usage-controller/internal/controller/controller.go:195 +0x2b
github.com/kyma-project/kyma/components/binding-usage-controller/internal/controller.(*ServiceBindingUsageController).(github.com/kyma-project/kyma/components/binding-usage-controller/internal/controller.worker)-fm()
	/home/jenkins/workspace/kyma/components/binding-usage-controller/src/github.com/kyma-project/kyma/components/binding-usage-controller/internal/controller/controller.go:191 +0x2a
github.com/kyma-project/kyma/components/binding-usage-controller/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc420347d78)
	/home/jenkins/workspace/kyma/components/binding-usage-controller/src/github.com/kyma-project/kyma/components/binding-usage-controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x54
github.com/kyma-project/kyma/components/binding-usage-controller/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc4203cbd78, 0x3b9aca00, 0x0, 0x1, 0xc420048660)
	/home/jenkins/workspace/kyma/components/binding-usage-controller/src/github.com/kyma-project/kyma/components/binding-usage-controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134 +0xbd
github.com/kyma-project/kyma/components/binding-usage-controller/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0xc420347d78, 0x3b9aca00, 0xc420048660)
	/home/jenkins/workspace/kyma/components/binding-usage-controller/src/github.com/kyma-project/kyma/components/binding-usage-controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
github.com/kyma-project/kyma/components/binding-usage-controller/internal/controller.(*ServiceBindingUsageController).Run(0xc4203e8000, 0xc420048660)
	/home/jenkins/workspace/kyma/components/binding-usage-controller/src/github.com/kyma-project/kyma/components/binding-usage-controller/internal/controller/controller.go:191 +0x20a
main.main()
	/home/jenkins/workspace/kyma/components/binding-usage-controller/src/github.com/kyma-project/kyma/components/binding-usage-controller/cmd/controller/main.go:120 +0xb29
```

**AC:**
- Bug is fixed 
- new validation is added to prevent similar errors in the future

**Comment by mszostok** [comment]
The new validator was added only for BUC project. Attempt to add it to other projects was too complex - fixing detected bugs by `errcheck` in Helm Broker takes over 1h and was not finished (see how much was done: https://github.com/kyma-project/kyma/compare/master...mszostok:fix-errcheck-all-projects?expand=1), because of that I decided to fix only BUC component as it was the main priority in this task.

Applying new `errcheck` validator for other projects needs to be covered in some follow-up task.
